httputil/client.go

```diff
@@ -30,3 +42,66 @@ func GetRequestIP(req *http.Request) string {
 
 	return strings.ReplaceAll(addresses[0], " ", "")
 }
+
+func DefaultTransport() *http.Transport {
+	tr := &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		ForceAttemptHTTP2:     DefaultForceHttp2,
+		MaxIdleConns:          DefaultMaxIdleConns,
+		IdleConnTimeout:       DefaultIdleConnTimeout,
+		TLSHandshakeTimeout:   DefaultTLSHandshakeTimeout,
+		ExpectContinueTimeout: DefaultExpectContinueTimeout,
+		DisableKeepAlives:     DefaultDisableKeepAlives,
+		MaxConnsPerHost:       DefaultMaxConnsPerHost,
+		MaxIdleConnsPerHost:   DefaultMaxIdleConnsPerHost,
+	}
+	return tr
+}
+
+type HttpClientOptions func(*http.Client)
+
+// DefaultHttpClient returns a default HTTP client with a custom transport configuration.
+// It disables keep-alives, sets max connections per host, and configures idle connection timeout.
+// This is useful for clients that need to make many short-lived requests without reusing connections.
+// It also sets a default timeout of 30 seconds.
+func DefaultHttpClient() *http.Client {
+	return &http.Client{
+		Transport: DefaultTransport(),
+		Timeout:   DefaultRequestTimeout,
+	}
+}
+
+// NewHttpClient creates a configured HTTP client with customizable options.
+// It initializes a client with DefaultHttpClient settings and applies functional options.
+// Parameters:
+//   - options: Variadic list of HttpClientOptions functions to customize client behavior
+//
+// Retur
+// Returns:
+//   - *http.Client configured with specified options
+func NewHttpClient(options ...HttpClientOptions) *http.Client {
+	client := DefaultHttpClient()
+	for _, option := range options {
+		option(client)
+	}
+	return client
+}
+
```

Comment:
It would be useful to have a `WithConfig(prefix string, conf config.Config)`


Similar to what we do for jobsDB:
* https://github.com/rudderlabs/rudder-server/blob/master/jobsdb/jobsdb.go#L711
* https://github.com/rudderlabs/rudder-server/blob/master/jobsdb/jobsdb.go#L992


It should support two levels:
* a global app level, env example: `RSERVER_HTTP_CLIENT_MAX_IDLE_CONNECTION` 
* and one with prefix (reporting), env example `RSERVER_HTTP_CLIENT_REPORTING_MAX_IDLE_CONNECTION`


This should give us the ability to modify any of those defaults during runtime with no additional boiler plate.

Reply:
I suggested it here via a new constructor but indeed with an option it makes more sense: https://github.com/rudderlabs/rudder-go-kit/pull/854#discussion_r2190280800